### DISCUSSION TITLE
Using dict for project default_options and target override_options

### DISCUSSION THREAD

**Discussion by bruchar1** [mesonbuild/meson#11285] [Ideas]
Is there any reason why meson do not accept options as a dict, in the form:

`project(options: {'key': 'value'})` instead of (or in addition to) `project(options: ['key=value']`?

It could also accept the following syntaxes:
- `{'subproject:key': 'value'}`
- `{subproject: {'key': 'value'}}`

**Comment by xclaesse** [comment]
There are no particular reason, PR welcome :)

**Reply by dcbaker** [reply]
Actually, there is. Before we assured that dictionaries were ordered they would have had surprising behavior. We do that now, so PR definitely welcome :)

**Reply by bruchar1** [reply]
I'm working on it...

**Reply by bruchar1** [reply]
@dcbaker I realise this interfers with the refactors your are doing about typed_kwargs. I'm probably better to wait you finish implementing it in the build process.

**Reply by bruchar1** [reply]
Implemented in https://github.com/mesonbuild/meson/pull/11493.
